conda/core/index.py

```diff
@@ -174,6 +174,8 @@ def _supplement_index_with_system(index):
     registered_names = []
     packages = context.plugin_manager.get_hook_results("virtual_packages")
     for package in packages:
+        if package.name is None:
```

Comment:
This is silencing a buggy virtual package instance, that doesn't have a name, I'd rather not swallow that here. We could raise a `PluginError` instead of course, which might be useful to help user identify where the problem is?

Reply:
Let's decide on how to handle such things when considering https://github.com/conda/conda/issues/12149 .

Reply:
Sounds good.